Userland/Libraries/LibWeb/WebIDL/AbstractOperations.cpp

```diff
@@ -301,4 +304,55 @@ JS::Completion construct(WebIDL::CallbackType& callback, JS::MarkedVector<JS::Va
     return completion;
 }
 
+// https://webidl.spec.whatwg.org/#dfn-named-property-visibility
+JS::ThrowCompletionOr<bool> is_named_property_exposed_on_object(Variant<Bindings::LegacyPlatformObject const*, HTML::Window const*> const& variant, JS::PropertyKey const& property_key)
```

Comment:
I'm not sold on using a variant here... it's possible a template with two allowed types would be cleaner/better performant.